tasks/broad/ImputationTasks.wdl

```diff
@@ -0,0 +1,761 @@
+    File ref_dict
+    String basename
+    Int disk_size = ceil(4*(size(vcf, "GiB") + size(vcf_index, "GiB"))) + 20
+    String gatk_docker
+  }
+  command <<<
+
+    ## update the header of the merged vcf
+    gatk UpdateVCFSequenceDictionary \
+    --source-dictionary ~{ref_dict} \
+    --output ~{basename}.vcf.gz \
+    --replace -V ~{vcf} \
+    --disable-sequence-dictionary-validation
+  >>>
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GiB"
+  }
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task RemoveSymbolicAlleles {
+  input {
+    File original_vcf
+    File original_vcf_index
+    String output_basename
+    Int disk_size = ceil(3*(size(original_vcf, "GiB") + size(original_vcf_index, "GiB")))
+    String gatk_docker
+  }
+  command {
+    gatk SelectVariants -V ~{original_vcf} -xl-select-type SYMBOLIC -O ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+}
+
+task SeparateMultiallelics {
+  input {
+    File original_vcf
+    File original_vcf_index
+    String output_basename
+    Int disk_size =  ceil(2*(size(original_vcf, "GiB") + size(original_vcf_index, "GiB")))
+    String bcftools_docker
+  }
+  command {
+    bcftools norm -m - ~{original_vcf} -Oz -o ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+}
+
+task OptionalQCSites {
+  input {
+    File input_vcf
+    File input_vcf_index
+    String output_vcf_basename
+    String bcftools_vcftools_docker
+  }
+  Int disk_size = ceil(2*(size(input_vcf, "GiB") + size(input_vcf_index, "GiB")))
+
+  command <<<
+    # site missing rate < 5% ; hwe p > 1e-6
+    vcftools --gzvcf ~{input_vcf}  --max-missing 0.05 --hwe 0.000001 --recode -c | bgzip -c > ~{output_vcf_basename}.vcf.gz
+    bcftools index -t ~{output_vcf_basename}.vcf.gz # Note: this is necessary because vcftools doesn't have a way to output a zipped vcf, nor a way to index one (hence needing to use bcf).
+  >>>
+
+  runtime {
+    docker: bcftools_vcftools_docker
+    memory: "16 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_vcf = "~{output_vcf_basename}.vcf.gz"
+    File output_vcf_index = "~{output_vcf_basename}.vcf.gz.tbi"
+  }
+}
+
+task SortIds {
+  input {
+    File vcf
+    String basename
+    Int disk_space =  3*ceil(size(vcf, "GiB"))
+    String bcftools_docker
+  }
+
+  command <<<
+    # what better way is there to do this I really don't know
```

Comment:
Not sorting the vcf, but sorting the alleles in the vcf ids.  The id's are first set to chr:pos:alt:ref, and then sorted so that the order of the alleles is lexicographical instead of alt:ref.  This is done with a mind towards prs scoring, to avoid ambiguity since prs weights files generally specify a1, a2, instead of alt:ref (it does add an insertion-deletion ambiguity, but that is much less likely to cause any problems).